lightning-invoice/src/de.rs

```diff
@@ -26,6 +27,98 @@ use super::{Bolt11Invoice, Sha256, TaggedField, ExpiryTime, MinFinalCltvExpiryDe
 
 use self::hrp_sm::parse_hrp;
 
+/// Trait for paring/converting base32 slice.
+pub trait FromBase32: Sized {
+	/// The associated error which can be returned from parsing (e.g. because of bad padding).
+	type Err;
+
+	/// Convert a base32 slice to `Self`.
+	fn from_base32(b32: &[Fe32]) -> Result<Self, Self::Err>;
```

Comment:
That would make sense, it would be more symmetric, and opens the room for more optimizations (less allocations).
It's not huge, but as it affects several `from_base32` implementations, I prefer to move it in a follwoup